File: customer-java-webflux/src/main/java/com/prez/db/CustomerPreferencesRepository.java
package com.prez.db;

import com.prez.model.CustomerPreferences;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import reactor.core.publisher.Flux;

public interface CustomerPreferencesRepository extends ReactiveMongoRepository<CustomerPreferences, String> {
    Flux<CustomerPreferences> findByCustomerId(String customerId);
}

File: customer-kotlin-webflux/src/main/kotlin/com/prez/db/CustomerPreferencesRepository.kt
package com.prez.db;

import com.prez.model.CustomerPreferences
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Flux

interface CustomerPreferencesRepository : ReactiveMongoRepository<CustomerPreferences, String> {
    fun findByCustomerId(customerId: String): Flux<CustomerPreferences>
}
